rnd/autogpt_server/Dockerfile.ws

```diff
@@ -0,0 +1,45 @@
+FROM python:3.11-slim-buster as server_base
+
+# Set environment variables
+ENV PYTHONDONTWRITEBYTECODE 1
+ENV PYTHONUNBUFFERED 1
+
+
+WORKDIR /app
+
+RUN apt-get update \
+    && apt-get install -y build-essential curl ffmpeg wget libcurl4-gnutls-dev libexpat1-dev gettext libz-dev libssl-dev \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/* \
+    && wget https://github.com/git/git/archive/v2.28.0.tar.gz -O git.tar.gz \
+    && tar -zxf git.tar.gz \
+    && cd git-* \
+    && make prefix=/usr all \
+    && make prefix=/usr install
+
+
+ENV POETRY_VERSION=1.8.3 \
+    POETRY_HOME="/opt/poetry" \
+    POETRY_NO_INTERACTION=1 \
+    POETRY_VIRTUALENVS_CREATE=false \
+    PATH="$POETRY_HOME/bin:$PATH"
+RUN pip3 install poetry
+
+COPY rnd/autogpt_server /app/rnd/autogpt_server
+COPY rnd/autogpt_libs /app/rnd/autogpt_libs
+COPY autogpt /app/autogpt
+COPY forge /app/forge
+
+WORKDIR /app/rnd/autogpt_server
+
+# Install dependencies
+RUN poetry install --no-interaction --no-ansi
+
+RUN poetry run prisma generate
```

Comment:
To reduce rebuild time and improve DX, could you reorder this so it doesn't have to re-install all dependencies when changing something in `rnd/autogpt_server`?

Reply:
1. Copy `pyproject.toml` and `poetry.lock`, and any path dependencies (so `forge` and `autogpt_libs`)
2. Install dependencies
3. Copy Prisma schema
4. `prisma generate`
5. Copy `rnd/autogpt_server`